### DISCUSSION TITLE
Implementation in React

### DISCUSSION THREAD

**Discussion by maacoun** [vagran/dxf-viewer#91] [Q&A] [answered]
### Implementation of dxf-viewer in React

Hello!

I hope you're doing well. I'm currently working on a React project that involves implementing DXF/DWG viewer. In my research, I've come across this js library, and it seems like a promising solution.

I'm reaching out to the community for guidance on effectively integrating the "dxf-viewer" library into React application. Specifically, I'm interested in understanding the steps for loading, parsing DXF files, and rendering the viewer components seamlessly within a React environment.

If anyone has experience using this library with React, I would greatly appreciate any advice, code snippets, or best practices you could share. Additionally, if there are potential challenges or considerations I should be aware of, your insights would be invaluable. I know it has sample application in vue, but I am kinda scared of vue xd.

Thank you in advance for your time and assistance!

**Comment by vagran** [comment, marked as answer]
Hello!
Actually, I do not have any experience with React, but as I know, people used the `dxf-viewer` in their React projects, it should be quite straight-forward if you are familiar with React.
Some brief description of how it should be done here: https://github.com/vagran/dxf-viewer-example-src/issues/12
Here is some code snippet from another issue: https://github.com/vagran/dxf-viewer-example-src/issues/13#issuecomment-1410231850 (I cannot comment on its idiomatic clearness since I am not familiar with React).
